exllamav2/generator/streaming.py

```diff
@@ -67,13 +67,17 @@ def __init__(self, model, cache, tokenizer, draft_model = None, draft_cache = No
 
     def set_stop_conditions(self, stop_conditions):
 
-        assert isinstance(stop_conditions, list)
+        assert isinstance(stop_conditions, (list, tuple, set))
 
-        self.stop_strings = []
-        self.stop_tokens = []
+        self.stop_strings = set()
+        self.stop_tokens = set()
         for t in stop_conditions:
-            if isinstance(t, int): self.stop_tokens += [t]
-            elif isinstance(t, str): self.stop_strings += [t]
+            if isinstance(t, int):
+              self.stop_tokens.add(t)
+            elif isinstance(t, str):
+              t_tokens = self.tokenizer.encode(t)[0]
+              if len(t_tokens) == 1: self.stop_tokens.add(t_tokens[0])
```

Comment:
I get the idea here, but there's no guarantee that the model will output the token corresponding to a stop string which encodes to a single token. E.g. if the stop string is `.` and the token picked is `...` that would fail to register as a stop condition.

Reply:
Right... let's just stick to sets then